### ISSUE TITLE
/v2/continents

### ISSUE THREAD

**Issue by shoeffner** [JuxhinDB/gw2-api-interface#7]
The /v2/continents endpoint is a little bit more difficult than the other endpoints and has a structure like this:

```
/v2/continents/<continentid>/floors/<floorid>/regions/<regionid>/maps/<mapid>
```
for example https://api.guildwars2.com/v2/continents/2/floors/1/regions/6/maps/795 would map to Legacy of the Foefire.

Each of these intermediate routes have a list available as well, so e.g. https://api.guildwars2.com/v2/continents/2/floors/1/regions/6/maps lists all sPvP maps while https://api.guildwars2.com/v2/continents/2/floors/1/regions/7/maps lists all WvW maps. https://api.guildwars2.com/v2/continents/2/floors/1/regions just lists 6 and 7 as valid regions, and so on.

The current implementation only allows to access the very first level, `/v2/continents/<continentid>`.

I am not sure how exactly the APIObject can be extended to give this functionality, except for following your usual approach, e.g. `.continentsfloors.get(id=2)`, but it seems difficult to resolve ambiguities here. An idea might be to use specific ids: `.continentsfloors.get(cid=2)` to get the list of floors and `.continentsfloors.get(cid=2, fid=1)` to get the specific floor.

While there is /maps inside the API, the continents API contains different information, e.g. waypoint locations etc.

Do you have good ideas on how to build it? I can try to build it and create a PR, if you would like me to.

**Comment by JuxhinDB** [comment]
@shoeffner -- I believe at some point I thought something like this might pop up, and should allow for you to specify a custom URL that overrides everything else as seen [here](https://github.com/JuxhinDB/gw2-api-interface/blob/master/gw2api/objects/base_object.py#L43)

Could you instead try `Continents.get(id='foo', floor='bar', region='baz', map='bing')`? You can then override [Continent](https://github.com/JuxhinDB/gw2-api-interface/blob/master/gw2api/objects/api_version_2/__init__.py#L175) to something like:

```python

class Continent(BaseAPIv2Object):

    def get(self, id, **kwargs):
        endpoint_url = self._build_endpoint_base_url()
        endpoint_url = "{base_url}/{version}/continents/{_id}".format(
                                   base_url=GuildWars2Client.BASE_URL,
                                   version=GuildWars2Client.VERSION,
                                   _id=kwargs.get('id'))

        floor = kwargs.get('floor')
        region = kwargs.get('region')
        map = kwargs.get('map')

        if floor:
           endpoint_url += '/floors/{}'.format(floor)
           if region:
               endpoint_url += '/regions/{}'.format(region)
               if map:
                   endpoint_url += '/maps/{}'.format(map)
 
       return super().get(url=endpoint_url)
```

That's handwritten without any testing so I'm pretty sure you need to tweak it around (it's ugly quite frankly). But I think it gets the idea across. The if-statements are cascading cause of their dependencies in the URL. 

If this makes sense to you, would you mind pushing a PR as I am kind of tight for the time being?

**Comment by shoeffner** [comment]
This approach will not allow to get /continents/2/floors without a floor ID (and subsequent parts behave the same way), which is also entirely possible (lists available floors).

I will think about something and create a PR soon!

**Comment by JuxhinDB** [comment]
@shoeffner - That makes sense; I would incorporate something in the core. However it would involve a fair amount of work at this point for an edge-case. I don't mind something somewhat hacky, especially if it's an edge case provided that it's documented well.

If you wish, you can use some magic value (e.g. `floor='all'`) to indicate that you want to include `/floors` but not `/floor/{id}`

**Comment by JuxhinDB** [comment]
Closing as has been fixed in PR #8
